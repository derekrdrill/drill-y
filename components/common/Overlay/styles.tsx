import styled from 'styled-components';
import { Grid } from '@mui/material';

export const OverlayRoot = styled(Grid)({
  backgroundColor: 'gray',
  height: '100%',
  left: 0,
  minHeight: '100%',
  minWidth: '100%',
  opacity: 0.6,
  position: 'fixed',
  top: 0,
  width: '100%',
});
