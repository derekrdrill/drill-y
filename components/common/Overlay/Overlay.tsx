import * as React from 'react';
import { GridProps } from '@mui/material';

import { OverlayRoot } from './styles';

export const getOverlayDisplay = (isOverlayVisible: boolean) =>
  isOverlayVisible ? 'block' : 'none';

export const getZIndexSmall = (isSidebarOpen: boolean) => (isSidebarOpen ? 4 : 1);
export const getZIndexLarge = (isQuickViewOpen: boolean) => (isQuickViewOpen ? 1 : 4);

interface OverlayProps extends GridProps {
  isQuickViewOpen: boolean;
  isSidebarOpen: boolean;
}

const Overlay = ({ isSidebarOpen, isQuickViewOpen }: OverlayProps) => (
  <OverlayRoot
    display={{
      xs: getOverlayDisplay(isSidebarOpen),
      lg: getOverlayDisplay(isQuickViewOpen),
    }}
    sx={{ zIndex: { xs: getZIndexSmall(isSidebarOpen), lg: getZIndexLarge(isQuickViewOpen) } }}
  />
);

export default Overlay;
