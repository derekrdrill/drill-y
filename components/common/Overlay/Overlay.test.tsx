import '@testing-library/jest-dom';
import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';

import Overlay, { getOverlayDisplay, getZIndexSmall, getZIndexLarge } from './Overlay';

describe('Overlay unit tests', () => {
  it('renders as expected', () => {
    const result = renderer
      .create(<Overlay isSidebarOpen={false} isQuickViewOpen={true} />)
      .toJSON();
    expect(result).toMatchSnapshot();
  });

  it('runs getOverlayDisplay correctly', () => {
    expect(getOverlayDisplay(true)).toEqual('block');
    expect(getOverlayDisplay(false)).toEqual('none');
  });

  it('runs getZIndexSmall correctly', () => {
    expect(getZIndexSmall(true)).toEqual(4);
    expect(getZIndexSmall(false)).toEqual(1);
  });

  it('runs getZIndexLarge correctly', () => {
    expect(getZIndexLarge(true)).toEqual(1);
    expect(getZIndexLarge(false)).toEqual(4);
  });
});
