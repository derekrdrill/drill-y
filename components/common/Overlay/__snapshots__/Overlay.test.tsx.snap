// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Overlay unit tests renders as expected 1`] = `
<div
  className="MuiGrid-root sc-beqWaB WkIvD css-1sm8red-MuiGrid-root"
/>
`;
